src/main/java/seedu/address/logic/parser/student/FindCommandParser.java

```diff
@@ -0,0 +1,76 @@
+package seedu.address.logic.parser.student;
+
+import static seedu.address.logic.parser.CliSyntax.PREFIX_ID;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_MODULE;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_NAME;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_TUTORIAL;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+import java.util.function.Predicate;
+
+import seedu.address.logic.commands.student.FindCommand;
+import seedu.address.logic.parser.ArgumentMultimap;
+import seedu.address.logic.parser.ArgumentTokenizer;
+import seedu.address.logic.parser.Parser;
+import seedu.address.logic.parser.ParserUtil;
+import seedu.address.logic.parser.Prefix;
+import seedu.address.logic.parser.exceptions.ParseException;
+import seedu.address.model.student.Student;
+import seedu.address.model.student.predicates.ModuleContainsKeywordPredicate;
+import seedu.address.model.student.predicates.NameContainsKeywordsPredicate;
+import seedu.address.model.student.predicates.StudentIdContainsKeywordPredicate;
+import seedu.address.model.student.predicates.TutorialContainsKeywordPredicate;
+
+/**
+ * Parses input arguments and creates a new FindCommand object
+ */
+public class FindCommandParser implements Parser<FindCommand> {
+
+    /**
+     * Parses the given {@code String} of arguments in the context of the FindCommand
+     * and returns a FindCommand object for execution.
+     * @throws ParseException if the user input does not conform the expected format
+     */
+    public FindCommand parse(String args) throws ParseException {
+        ArgumentMultimap argMultimap =
+                ArgumentTokenizer.tokenize(args, PREFIX_NAME, PREFIX_ID, PREFIX_MODULE, PREFIX_TUTORIAL);
+
+        ParserUtil.assertAnyPrefixesPresent(argMultimap,
+                PREFIX_NAME, PREFIX_ID, PREFIX_MODULE, PREFIX_TUTORIAL);
+
+        List<Predicate<Student>> predicates = new ArrayList<>();
+
+        Prefix[] prefixes = {PREFIX_NAME, PREFIX_ID, PREFIX_MODULE, PREFIX_TUTORIAL};
+        for (Prefix prefix : prefixes) {
+            Optional<String> str = argMultimap.getValue(prefix);
+            if (str.isEmpty()) {
+                continue;
+            }
+            String trimmedKeywords = str.get().trim();
```

Comment:
I just want to point out that the current find is not case-sensitive. So when I try `find n/Yu` and `find n/yu` it gives me the same result. I'm not sure if it's intentional or not. If you expect it to behave these ways, just ignore my comment.

Reply:
Yes, we should keep it and explicit state that it is case-insensitive. It is more user-friendly